concent_api/concent_api/middleware.py

```diff
@@ -39,3 +49,81 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+def should_use_html(request):
+    """
+    Function inspects request headers and determines is response should be in HTML format.
+    1) By default HTML should not be used (JSON is the default format)
+    2) If `text/html` is present in 'Accept' header and 'application/json' is not, HTML should be used
+    3) If both are present, HTML will be used if it has greater weight than JSON
+    """
+    def get_weight(selected_content_type, accepted_content_types):
+        for accepted_content_type in accepted_content_types.split(','):
+            if selected_content_type in accepted_content_type:
+                match = re.search(f"(?:{selected_content_type})(;q=(?P<weight>0.\d))?", accepted_content_type)
+                if match is not None and match.group('weight') is not None:
```

Comment:
I don't think `(?P<weight>0.\d)` is correct:
- `.` will match any character, not just a dot. Should be `\.`.
- It won't match `q=1` and `q=0`.

BTW, here's the spec for the weight: https://tools.ietf.org/html/rfc7231#section-5.3.1

> ```
>      weight = OWS ";" OWS "q=" qvalue
>      qvalue = ( "0" [ "." 0*3DIGIT ] )
>             / ( "1" [ "." 0*3("0") ] )
> ```